azuread/resource_application.go

```diff
@@ -517,3 +590,75 @@ func flattenADApplicationOauth2Permissions(in *[]graphrbac.OAuth2Permission) []m
 
 	return result
 }
+
+func expandADApplicationAppRoles(input []interface{}) *[]graphrbac.AppRole {
```

Comment:
Then this becomes:
```suggestion
func expandADApplicationAppRoles(i interface{}) *[]graphrbac.AppRole {
	input := i.(*schema.Set).List()
```

Reply:
thanks for your help with this, I have updated with your suggestions